src/app/sv_global_parameter.rs

```diff
@@ -0,0 +1,34 @@
+use crate::*;
+
+mod impl_ {
+    use crate::*;
+
+    extern "C" {
+        #[link_name = "_ZN3app19sv_global_parameter28melee_melee_info_item_appearEv"]
+        pub(super) fn melee_melee_info_item_appear() -> phx::Hash40;
+
+        #[link_name = "_ZN3app19sv_global_parameter15melee_rule_modeEv"]
+        pub(super) fn melee_rule_mode() -> phx::Hash40;
+
+        #[link_name = "_ZN3app19sv_global_parameter15melee_rule_typeEv"]
+        pub(super) fn melee_rule_type() -> phx::Hash40;
+    }
+}
+
+pub fn melee_melee_info_item_appear() -> phx::Hash40 {
```

Comment:
If you are interested, I would not be opposed to you translating these into an enum. I know that you were talking in SUM about these being essentially scoped hashes (*shudders*) so if you wanted to turn them into an enum I would be more than supportive of that. Your call ultimately.

Reply:
I would prefer to keep the function return value and type unchanged.